lib/mixlib/versioning/format/rubygems.rb

```diff
@@ -41,23 +45,56 @@ class Format
       #
       # @author Seth Chisamore (<schisamo@chef.io>)
       # @author Christopher Maier (<cm@chef.io>)
+      # @author Ryan Hass (<rhass@chef.io>)
       class Rubygems < Format
-        RUBYGEMS_REGEX = /^(\d+)\.(\d+)\.(\d+)(?:\.([[:alnum:]]+(?:\.[[:alnum:]]+)?))?(?:\-(\d+))?$/
-
         # @see Format#parse
         def parse(version_string)
-          match = version_string.match(RUBYGEMS_REGEX) rescue nil
+          # Pure regex along will never match partial versions correctly in the
+          # expected groupings. This is becausel partial versions shift the
+          # grouping values and break the mappings to for each appropriate value.
+          rubygems_version = version_string.scan(/[0-9]+|[a-z]+/i).map do |s|
+            /^\d+$/ =~ s ? s.to_i : s
+          end rescue nil
+
+          # Each value for MAJOR.MINOR.PATCH must be a integer.
+          unless rubygems_version && rubygems_version[0..2].map {|v| v.is_a?(Integer) }.all?
+            raise_invalid_version(version_string)
+          end
 
-          unless match
-            raise Mixlib::Versioning::ParseError, "'#{version_string}' is not a valid #{self.class} version string!"
+          if version_string =~ /\-\D+/
```

Comment:
Per comment above... execute this first and it should simply the `unless #all?` condition.